lib/helpers/query/castUpdate.js

```diff
@@ -42,6 +41,11 @@ module.exports = function castUpdate(schema, obj, options, context, filter) {
     return obj;
   }
 
+  if (schema.options.strict === 'throw') {
+    throw new StrictModeError();
```

Comment:
This should have an error message. You also need to check that `obj` actually contains the discriminator key